internal/tablemap.go

```diff
@@ -127,3 +127,43 @@ func GetSpannerCols(conv *Conv, srcTable string, srcCols []string) ([]string, er
 	}
 	return spCols, nil
 }
+
+// GetSpannerKeyName maps source foreign key name to
+// legal Spanner foreign key name
+// If the srcKeyName is empty string we can just return
+// empty string without error.
+// If the srcKeyName is not empty we need to make sure
+// of the following things:
+// a) the new foreign key name is legal
+// b) the new foreign key name doesn't clash with other Spanner
+//    foreignkey names
+// c) we consistently return the same key name.
+func GetSpannerKeyName(conv *Conv, srcKeyName string, schemaForeignKeys map[string]bool) (string, error) {
+	if srcKeyName == "" {
+		return "", nil
+	}
+	spKeyName, _ := FixName(srcKeyName)
+
+	// spKeyName has been used before i.e. FixName caused a collision.
+	// Add unique postfix: use number of foreign keys so far.
+	// However, there is a chance this has already been used,
+	// so need to iterate.
+	if _, found := schemaForeignKeys[spKeyName]; found {
+		id := len(schemaForeignKeys)
+		for {
+			c := spKeyName + "_" + strconv.Itoa(id)
+
+			// check wether the spKeyName is used before
```

Comment:
Suggest deleting this comment -- it is not quite accurate (we are checking whether c has been used before, not whether spKeyName has been used before), and I don't think it is needed, given the comment a few lines earlier.

Also, note that that go comments should be full sentences, and start with a capital (when appropriate) and end with a full stop. In general, go comments are used sparingly -- they are only used when there is something potentially confusing or surprising going on.

Similarly, blank lines are used sparingly -- see existing code for examples.

Reply:
sure nevin, i'll take care of comments, updated